File: src/main/java/hu/frontrider/arcana/sided/client/gui/GuiExperimentTableCage.java
package hu.frontrider.arcana.sided.client.gui;

import hu.frontrider.arcana.content.blocks.experiments.tiles.TileEntityExperimentTable;
import net.minecraft.block.Block;
import net.minecraft.client.gui.inventory.GuiContainer;
import net.minecraft.client.renderer.GlStateManager;
import net.minecraft.entity.player.InventoryPlayer;
import net.minecraft.inventory.Container;
import net.minecraft.util.ResourceLocation;
import net.minecraftforge.fml.common.registry.GameRegistry;

import static hu.frontrider.arcana.ThaumicArcana.MODID;

public class GuiExperimentTableCage extends GuiContainer {
    @GameRegistry.ObjectHolder(MODID + ":experiment_table")
    static Block experimentTable = null;

    private static final ResourceLocation BG_TEXTURE = new ResourceLocation(MODID + ":textures/gui/experiment_table_cage.png");
    private InventoryPlayer playerInv;
    private final TileEntityExperimentTable experimentTableTileEntity;

    public GuiExperimentTableCage(Container container, InventoryPlayer playerInv, TileEntityExperimentTable experimentTableTileEntity) {
        super(container);
        this.playerInv = playerInv;
        this.experimentTableTileEntity = experimentTableTileEntity;
    }

    @Override
    protected void drawGuiContainerBackgroundLayer(float partialTicks, int mouseX, int mouseY) {
        this.drawDefaultBackground();
        GlStateManager.color(1, 1, 1, 1);
        mc.getTextureManager().bindTexture(BG_TEXTURE);
        int x = (width - xSize) / 2;
        int y = (height - ySize) / 2;
        drawTexturedModalRect(x, y, 0, 0, xSize, ySize);
    }

    @Override
    public void drawScreen(int mouseX, int mouseY, float partialTicks) {
        super.drawScreen(mouseX, mouseY, partialTicks);
        this.renderHoveredToolTip(mouseX, mouseY);
    }

    @Override
    protected void drawGuiContainerForegroundLayer(int mouseX, int mouseY) {
        super.drawGuiContainerForegroundLayer(mouseX, mouseY);
    }
}

File: src/main/java/hu/frontrider/arcana/sided/client/gui/GuiExperimentTableCage.kt
package hu.frontrider.arcana.sided.client.gui

import hu.frontrider.arcana.content.blocks.experiments.tiles.TileEntityExperimentTable
import net.minecraft.block.Block
import net.minecraft.client.gui.inventory.GuiContainer
import net.minecraft.client.renderer.GlStateManager
import net.minecraft.entity.player.InventoryPlayer
import net.minecraft.inventory.Container
import net.minecraft.util.ResourceLocation
import net.minecraftforge.fml.common.registry.GameRegistry
import net.minecraftforge.fml.relauncher.Side
import net.minecraftforge.fml.relauncher.SideOnly

import hu.frontrider.arcana.ThaumicArcana.MODID

@SideOnly(Side.CLIENT)
class GuiExperimentTableCage(container: Container, private val playerInv: InventoryPlayer, private val experimentTableTileEntity: TileEntityExperimentTable) : GuiContainer(container) {

    override fun drawGuiContainerBackgroundLayer(partialTicks: Float, mouseX: Int, mouseY: Int) {
        this.drawDefaultBackground()
        GlStateManager.color(1f, 1f, 1f, 1f)
        mc.textureManager.bindTexture(BG_TEXTURE)
        val x = (width - xSize) / 2
        val y = (height - ySize) / 2
        drawTexturedModalRect(x, y, 0, 0, xSize, ySize)
    }

    override fun drawScreen(mouseX: Int, mouseY: Int, partialTicks: Float) {
        super.drawScreen(mouseX, mouseY, partialTicks)
        this.renderHoveredToolTip(mouseX, mouseY)
    }

    companion object {
        private val BG_TEXTURE = ResourceLocation("$MODID:textures/gui/experiment_table_cage.png")
    }
}
